src/mtx/commands/downlink/GetHalfHours.ts

```diff
@@ -0,0 +1,103 @@
+import Command, {TCommandExampleList, COMMAND_HEADER_SIZE} from '../../Command.js';
+import CommandBinaryBuffer from '../../CommandBinaryBuffer.js';
+import {DOWNLINK} from '../../../constants/directions.js';
+import {READ_ONLY} from '../../constants/accessLevels.js';
+import {IDate} from '../../utils/dateTime.js';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import GetHalfHoursResponse from '../uplink/GetHalfHoursResponse.js';
+
+interface IGetHalfHoursParameters {
+    date: IDate
+}
+
+
+const COMMAND_ID = 0x15;
+const COMMAND_SIZE = 3;
+
+
+const examples: TCommandExampleList = [
+    {
+        name: 'request half hour data for',
+        parameters: {
+            date: {
+                year: 24,
+                month: 3,
+                date: 22
+            }
+        },
+        hex: {header: '15 03', body: '18 03 16'}
+    }
+];
+
+
+/**
+ * Downlink command to get active A+ energy in half hours by date.
+ *
+ * The corresponding uplink command: {@link GetHalfHoursResponse}.
+ *
+ * @example
+ * ```js
+ * import GetHalfHours from 'jooby-codec/mtx/commands/downlink/GetHalfHours.js';
+ *
+ * const command = new GetHalfHours({
+ *     date: {
+ *         year: 24,
+ *         month: 3,
+ *         date: 22,
+ *     }
+ * });
+ *
+ * // output command binary in hex representation
+ * console.log(command.toHex());
+ * // 15 03 18 03 16
+ * ```
+ *
+ * [Command format documentation](https://github.com/jooby-dev/jooby-docs/blob/main/docs/mtx/commands/GetHalfHours.md#request)
+ */
+class GetHalfHours extends Command {
+    constructor ( public parameters: IGetHalfHoursParameters ) {
```

Comment:
do we need this additional level?


```suggestion
    constructor ( public parameters: IDate ) {
```
?

Reply:
fixed